react/CategoryPanel.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component, Fragment } from 'react'
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+import { Link, withRuntimeContext } from 'vtex.render-runtime'
+import { facetOptionShape } from './constants/propTypes'
+import searchResult from './searchResult.css'
+import CategoriesHighlights from 'vtex.store-components/CategoriesHighlights'
+
+class CategoryPanel extends Component {
+  static propTypes = {
+    /** Category tree */
+    tree: PropTypes.arrayOf(facetOptionShape),
+    /** Number of category filters to be shown */
+    noOfCategories: PropTypes.number,
+    /** Number of categories per row */
+    quantityOfItemsPerRow: PropTypes.number,
+  }
+
+  static defaultProps = {
+    tree: [],
+    noOfCategories: 1000,
+    quantityOfItemsPerRow: 4,
+  }
+
+  renderCategoryShelf = (
+    category,
+    noOfChildren,
+    quantityOfItemsPerRow,
+    mobile = false
+  ) => {
+    const categoriesHighlighted = category.Children.slice(0, noOfChildren).map(
+      child => ({
+        id: child.Id,
+        name: child.Name,
+        to: child.Link,
+      })
+    )
+    const headerClasses = classNames(
+      searchResult.categoryPanelHeaderRow,
+      'flex flex-row flex-wrap items-stretch pa3 bn',
+      {
+        mh4: !mobile,
+      }
+    )
+    const itemClasses = classNames(
+      searchResult.categoryPanelItemRow,
+      'flex flex-row flex-wrap items-stretch pa3 bn',
+      {
+        mh4: !mobile,
+      }
+    )
+    let i = 0
+    const items = []
+    for (i = 0; i < categoriesHighlighted.length; i += quantityOfItemsPerRow) {
+      items.push(categoriesHighlighted.slice(i, i + quantityOfItemsPerRow))
+    }
+    return (
+      <Fragment key={`parent-fragment-${category.Id}`}>
+        <div className={headerClasses}>
```

Comment:
I think there's no need for this div here, it could be the tag `<h3>` directly, since it's a block tag.

Reply:
I thought to have that `<div>` just keep the HTML structure cleaner (the generated rows below the heading have `<div>`s in the same way) and have more control over the styling there.

Reply:
Okay, I was late to figure this point out. It's done. Thanks man! :+1: